devices/snitch_cluster/trampoline_snitchCluster.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2024 ETH Zurich and University of Bologna.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Philip Wiese <wiesep@iis.ee.ethz.ch>
+
+#ifndef _CLUSTER_SNITCH_TRAMPOLINE_INCLUDE_GUARD_
+#define _CLUSTER_SNITCH_TRAMPOLINE_INCLUDE_GUARD_
+
+#include <stdint.h>
+
+/** \addtogroup devices_snitchCluster
+ *  @{
+ */
+
+/**
+ * @brief Setup the core to a known state.
+ *
+ * This function will set up the global pointer and thread pointer for the core.
+ */
+#define SNITCH_SETUP_CORE() \
+    asm volatile(".option push\n" \
+                 ".option norelax\n" \
+                 "la gp, __global_pointer$\n" \
+                 ".option pop\n" \
+                 "mv tp, zero\n" \
+                 : /* No outputs */ \
+                 : /* No inputs */ \
+                 : /* No clobbered registers */)
+
+/** @} */
```

Comment:
Why make this a macro and in a header file? This is clearly a function (i.e. emits program code) and the "natural expectation" is for functions to live in `.c` files. But maybe I am missing something important here.

Reply:
Well, in this context I think this is a reasonable way to go about it.
Can we rename the macro though? Something like "_SETUP_GP_TP" might be more descriptive.